core/description/resource.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package description
+
+import (
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+)
+
+// Resource represents an application resource.
+type Resource interface {
+	Name() string
+	Revision() int
+	CharmStoreRevision() int
+	AddRevision(ResourceRevisionArgs)
+	Revisions() map[int]ResourceRevision
+	Validate() error
+}
+
+// ResourceRevision represents a revision of an application resource.
+type ResourceRevision interface {
+	Revision() int
+	Type() string
+	Path() string
+	Description() string
+	Origin() string
+	FingerprintHex() string
```

Comment:
I did this because on resource.Resource the fingerprint is stored as a raw `[]byte` but here we convert it to a hex representation of that. The name is a reminder that a conversion has been performed.